airbyte-commons/src/main/java/io/airbyte/commons/logging/MaskedDataInterceptor.java

```diff
@@ -0,0 +1,138 @@
+    this.pattern = buildPattern(specMaskFile);
+  }
+
+  @Override
+  public LogEvent rewrite(final LogEvent source) {
+    return Log4jLogEvent.newBuilder()
+        .setLoggerName(source.getLoggerName())
+        .setMarker(source.getMarker())
+        .setLoggerFqcn(source.getLoggerFqcn())
+        .setLevel(source.getLevel())
+        .setMessage(new SimpleMessage(applyMask(source.getMessage().getFormattedMessage())))
+        .setThrown(source.getThrown())
+        .setContextMap(source.getContextMap())
+        .setContextStack(source.getContextStack())
+        .setThreadName(source.getThreadName())
+        .setSource(source.getSource())
+        .setTimeMillis(source.getTimeMillis())
+        .build();
+  }
+
+  /**
+   * Applies the mask to the message, if necessary.
+   *
+   * @param message The log message.
+   * @return The possibly masked log message.
+   */
+  private String applyMask(final String message) {
+    if (pattern.isPresent()) {
+      return message.replaceAll(pattern.get(), "\"$1\":\"" + MASK + "\"");
+    } else {
+      return message;
+    }
+  }
+
+  /**
+   * Loads the maskable properties from the provided file.
+   *
+   * @param specMaskFile The spec mask file.
+   * @return The set of maskable properties.
+   */
+  private Set<String> getMaskableProperties(final String specMaskFile) {
+    logger.info("Loading mask data from '{}", specMaskFile);
+    try {
+      final String maskFileContents = IOUtils.toString(getClass().getResourceAsStream(specMaskFile), Charset.defaultCharset());
+      final Map<String, Set<String>> properties = Jsons.object(Yamls.deserialize(maskFileContents), new TypeReference<>() {});
+      return properties.getOrDefault("properties", Set.of());
+    } catch (final Exception e) {
+      logger.error("Unable to load mask data from '{}': {}.", specMaskFile, e.getMessage());
+      return Set.of();
+    }
+  }
+
+  /**
+   * Builds the maskable property matching pattern.
+   *
+   * @param specMaskFile The spec mask file.
+   * @return The regular expression pattern used to find maskable properties.
+   */
+  private Optional<String> buildPattern(final String specMaskFile) {
+    final Set<String> maskableProperties = getMaskableProperties(specMaskFile);
+    return !maskableProperties.isEmpty() ? Optional.of(generatePattern(maskableProperties)) : Optional.empty();
+  }
+
+  /**
+   * Generates the property matching pattern string from the provided set of properties.
+   *
+   * @param properties The set of properties to match.
+   * @return The generated regular expression pattern used to match the maskable properties.
+   */
+  private String generatePattern(final Set<String> properties) {
+    final StringBuilder builder = new StringBuilder();
+    builder.append("(?i)"); // case insensitive
+    builder.append("\"(");
+    builder.append(properties.stream().collect(Collectors.joining("|")));
+    builder.append(")\"\\s*:\\s*\"?((\\\\\"|[^\",}])*)\"?");
```

Comment:
It may crash earlier in our pipeline if this happens, but I think that if my secret contains a quote, I think this would not replace everything.

Reply:
@gosusnp Good catch.  I'll add some tests to see what happens with a quote in the value of the property.

Reply:
I've added a test with a value with quotes and the regex finds and replaces it appropriately.  I agree that it seems like it shouldn't, but I think what is happening is the escape sequence for the quote allows it to match.